code/weapon/weapons.cpp

```diff
@@ -7497,7 +7524,55 @@ void weapon_hit( object * weapon_obj, object * other_obj, vec3d * hitpos, int qu
 		weapon_hit_do_sound(other_obj, wip, hitpos, armed_weapon, quadrant);
 	}
 
-	if (wip->impact_weapon_expl_effect.isValid() && armed_weapon) {
+
+ 	bool valid_conditional_impact = false;
+	int relevant_armor_idx = -1;
+	float relevant_fraction = 1.0f;
+	float hit_angle = 0.0f;
+	vec3d reverse_incoming = (vmd_zero_vector - weapon_obj->orient.vec.fvec);
+
+	if (hitnormal) {
+		hit_angle = vm_vec_delta_ang(hitnormal, &reverse_incoming, nullptr);
+	}
+
+	if (other_obj != nullptr && other_obj->type == OBJ_SHIP) {
+		shipp = &Ships[other_obj->instance];
+		if (quadrant == -1) {
+			relevant_armor_idx = shipp->armor_type_idx;
+			relevant_fraction = other_obj->hull_strength / shipp->ship_max_hull_strength;
+		} else {
+			relevant_armor_idx = shipp->shield_armor_type_idx;
+			relevant_fraction = ship_quadrant_shield_strength(other_obj, quadrant);
+		}
+		if (wip->conditional_impacts.count(relevant_armor_idx) == 1) {
+			const auto* cis = &wip->conditional_impacts[relevant_armor_idx];
+			for (auto &ci : *cis) {
```

Comment:
So I didn't mean a pointer.  Not that this code is particularly complicated, but when you type `const type*` only the pointer and not the contents are const.

But actually, if we're not going to use cis except as a bound for ci it might be best to do...

`for (const auto& ci : wip->conditional_impacts[relevant_armor_idx])